### PR TITLE
Add GN flag to disable building cloud print and disable pdfium on non-windows

### PR DESCRIPTION
**Pull request by rht**
This partially addresses #1958 except on Windows.
cc: @fmarier 

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions.

### CODE REVIEW

File: lib/config.js [hunk 1]
```diff
@@ -91,7 +91,9 @@ Config.prototype.buildArgs = function () {
     chrome_version_major: chrome_version_parts[0],
     safebrowsing_api_endpoint: this.safeBrowsingApiEndpoint,
     brave_referrals_api_key: this.braveReferralsApiKey,
-    enable_hangout_services_extension: this.enable_hangout_services_extension
+    enable_hangout_services_extension: this.enable_hangout_services_extension,
+    enable_printing: false,  // disable cloudprint
```
Comment (bridiver) [inline lib/config.js]: is this just cloud printing? cc @darkdh

Comment (rht) [reply lib/config.js]: Whoops, that's all of printing. I found the flag at https://github.com/Eloston/ungoogled-chromium/blob/master/patches/inox-patchset/0005-disable-default-extensions.patch#L19-L23. But it looks like these particular lines need to be removed in order to disable just cloud printing.

**Comment by rht** [comment]
From the output of https://github.com/brave/brave-core/pull/980#issuecomment-453213508, technically `enable_print_preview` is technically `true` for all platforms. So pdfium can't be disabled.

**Comment by fmarier** [comment]
This latest patch runs into the same assertion failure as before:

```
ERROR at //printing/BUILD.gn:20:3: Assertion failed.
  assert(enable_pdf,
  ^-----
Windows basic printing or print preview needs pdf: set enable_pdf=true.
See //BUILD.gn:142:7: which caused the file to be included.
      "//printing:printing_unittests",
      ^------------------------------
```

**Comment by rht** [comment]
Yeah, it is impossible to disable unless print preview uses pdf.js.

**Comment by bsclifton** [comment]
Closing as we solved this issue and are now using PDFium

**Comment by rht** [comment]
https://github.com/brave/brave-browser/issues/3846

**Pull request closed without merging**